core/src/main/java/hudson/model/Computer.java

```diff
@@ -669,41 +672,59 @@ public boolean isLaunchSupported() {
     @Exported
     @Deprecated
     public boolean isTemporarilyOffline() {
-        return temporarilyOffline;
+        var node = getNode();
+        return node != null && node.isTemporarilyOffline();
     }
 
     /**
      * @deprecated as of 1.320.
-     *      Use {@link #setTemporarilyOffline(boolean, OfflineCause)}
+     *      Use {@link #setTemporaryOfflineCause(OfflineCause)}
      */
     @Deprecated
     public void setTemporarilyOffline(boolean temporarilyOffline) {
-        setTemporarilyOffline(temporarilyOffline, null);
+        setTemporaryOfflineCause(temporarilyOffline ? new OfflineCause.LegacyOfflineCause() : null);
+    }
+
+    /**
+     * @deprecated as of TODO.
+     *      Use {@link #setTemporaryOfflineCause(OfflineCause)} instead.
+     */
+    @Deprecated
+    public void setTemporarilyOffline(boolean temporarilyOffline, OfflineCause cause) {
```

Comment:
The behaviour of this method changed. 
Previously if you set the `temporarilyOffline` to false, this ensured that the cause was ignored.
In the vsphere-cloud plugin they call this method with `false` and a cause.
https://github.com/jenkinsci/vsphere-cloud-plugin/blob/26db2ff0f928c0013bb1f448d65f933f61816efa/src/main/java/org/jenkinsci/plugins/vSphereCloudSlave.java#L321
Of course this is a bug in the plugin one can say

Reply:
Fixed